$MirOS: ports/databases/sqlite/patches/patch-configure_ac,v 1.5 2007/05/06 20:22:31 tg Exp $
--- configure.ac.orig	Tue Aug  5 19:58:25 2008
+++ configure.ac	Tue Aug 26 12:17:08 2008
@@ -326,7 +326,10 @@ else
   AC_MSG_RESULT(unknown)
 fi
 if test "$CYGWIN" != "yes"; then
-  AC_CYGWIN
+	case $host_os in
+	*cygwin*)	CYGWIN=yes;;
+	*)		CYGWIN=no;;
+	esac
 fi
 if test "$CYGWIN" = "yes"; then
   BUILD_EXEEXT=.exe
@@ -531,8 +534,10 @@ if test x"$with_readline" != xno; then
 	if test "x$with_readline_lib" = xauto; then
 		save_LIBS="$LIBS"
 		LIBS=""
-		AC_SEARCH_LIBS(tgetent, [readline ncurses curses termcap], [term_LIBS="$LIBS"], [term_LIBS=""])
+		AC_SEARCH_LIBS(tgetent, [edit readline ncurses curses termcap], [term_LIBS="$LIBS"], [term_LIBS=""])
+		AC_CHECK_LIB([edit], [readline], [TARGET_READLINE_LIBS="-ledit"],
 		AC_CHECK_LIB([readline], [readline], [TARGET_READLINE_LIBS="-lreadline"], [found="no"])
+		)
 		TARGET_READLINE_LIBS="$TARGET_READLINE_LIBS $term_LIBS"
 		LIBS="$save_LIBS"
 	else
